pkg/controller/template/render.go

```diff
@@ -128,7 +128,7 @@ func platformFromControllerConfigSpec(ic *mcfgv1.ControllerConfigSpec) (string,
 		return "", fmt.Errorf("cannot generate MachineConfigs when no platform is set")
 	case platformBase:
 		return "", fmt.Errorf("platform _base unsupported")
-	case platformAWS, platformAzure, platformBaremetal, platformGCP, platformOpenstack, platformLibvirt, platformNone:
+	case platformAWS, platformAzure, platformBaremetal, platformGCP, platformOpenstack, platformLibvirt, platformNone, platformVSphere:
```

Comment:
cc @eparis @staebler is the cloud provider ready to be turned on the kubelet.
in previous conversation, setting vsphere for kubelet caused failures.

Reply:
The last that I saw the kubelet would crash loop with the cloud provider set. I believe that the contention from @eparis was that the crash loop was at least partially due to inappropriate permissions when accessing the creds secret. I do not know of any work to resolve that.

Reply:
Is there an issue that I can link to in a comment and in the issue?

Reply:
@eparis @abhinavdahiya @staebler would be nice to understand if this is an issue with 4.2 or it's _still_ the expected behavior. The BZ has been closed based on @staebler's comment above but happy to bring it back and merge this if this is 4.2